### PR TITLE
Update golint keys

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**

### CODE REVIEW

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 1]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("PackageComment", msg ->
+      msg.startsWith("package comment should be of the form") ||
+        msg.startsWith("package comment should not have leading space") ||
+        msg.equals("package comment is detached; there should be no blank lines between it and the package statement") ||
+        msg.equals("should have a package comment, unless it's in another file for this package")),
+    new ExternalKey("BlankImports", msg -> msg.equals("a blank import should be only in a main or test package, or have a comment justifying it")),
+    new ExternalKey("Imports", msg -> msg.equals("should not use dot imports")),
+    new ExternalKey("Exported", msg -> msg.matches("exported [a-zA-Z0-9\\.]+ [a-zA-Z0-9\\.]+ should have comment[\\(\\) a-zA-Z0-9]* or be unexported") ||
+      msg.matches("comment on exported type [a-zA-Z0-9]+ should be of the form \"[a-zA-Z0-9]+ ...\" \\\\(with optional leading article\\\\)") ||
+      msg.startsWith("comment on exported") ||
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource previous `matches` can be dropped

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 2]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+      msg.endsWith("should have its own declaration") ||
       msg.matches("[a-zA-Z0-9]+ name will be used as [a-zA-Z0-9]+.[a-zA-Z0-9]+ by other packages, and that stutters; consider calling this [a-zA-Z0-9]+")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource how about replacing this `matches` with `contains`?

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 3]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
-    new ExternalKey("UnderscoreInPackageName", msg -> msg.startsWith("don't use an underscore in package name"))
+    new ExternalKey("Names", msg ->
+      msg.startsWith("don't use an underscore in package name") ||
+        msg.startsWith("don't use ALL_CAPS in Go names; use CamelCase") ||
+        msg.startsWith("don't use leading k in Go names;") ||
+        msg.startsWith("don't use underscores in Go names;") ||
+        msg.matches("(range var|struct field|[a-zA-Z0-9]+) [a-zA-Z0-9]+ should be [a-zA-Z0-9]+") ||
+        msg.startsWith("don't use MixedCaps in package name;")
+      ),
+    new ExternalKey("VarDecls", msg -> (msg.contains("from declaration of var") && msg.endsWith("it is the zero value")) ||
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource I think we can just keep `msg.contains("from declaration of var")` for this key

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 4]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+      (msg.startsWith("should omit type") && msg.endsWith("it will be inferred from the right-hand side"))),
+    new ExternalKey("Elses", msg -> msg.contains("if block ends with a return statement, so drop this else and outdent its block")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource `startsWith`

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 5]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("Ranges", msg -> msg.matches("should omit (2nd value|values) from range; this loop is equivalent to .*")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource can be simplified `contains("from range; this loop is equivalent to")`

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 6]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
-    new ExternalKey("PackageCommentDetached", msg -> msg.equals("package comment is detached; there should be no blank lines between it and the package statement")),
-    new ExternalKey("PackageNames", msg ->
+    new ExternalKey("PackageComment", msg ->
+      msg.startsWith("package comment should be of the form") ||
+        msg.startsWith("package comment should not have leading space") ||
+        msg.equals("package comment is detached; there should be no blank lines between it and the package statement") ||
+        msg.equals("should have a package comment, unless it's in another file for this package")),
+    new ExternalKey("BlankImports", msg -> msg.equals("a blank import should be only in a main or test package, or have a comment justifying it")),
+    new ExternalKey("Imports", msg -> msg.equals("should not use dot imports")),
+    new ExternalKey("Exported", msg -> msg.matches("exported [a-zA-Z0-9\\.]+ [a-zA-Z0-9\\.]+ should have comment[\\(\\) a-zA-Z0-9]* or be unexported") ||
```
Comment (alban-auzeill) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: You should include '_' and replace [a-zA-Z0-9\\.] by [\\w\\.]

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 7]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+        msg.startsWith("don't use leading k in Go names;") ||
+        msg.startsWith("don't use underscores in Go names;") ||
+        msg.matches("(range var|struct field|[a-zA-Z0-9]+) [a-zA-Z0-9]+ should be [a-zA-Z0-9]+") ||
+        msg.startsWith("don't use MixedCaps in package name;")
+      ),
+    new ExternalKey("VarDecls", msg -> (msg.contains("from declaration of var") && msg.endsWith("it is the zero value")) ||
+      (msg.startsWith("should omit type") && msg.endsWith("it will be inferred from the right-hand side"))),
+    new ExternalKey("Elses", msg -> msg.contains("if block ends with a return statement, so drop this else and outdent its block")),
+    new ExternalKey("Ranges", msg -> msg.matches("should omit (2nd value|values) from range; this loop is equivalent to .*")),
+    new ExternalKey("Errorf", msg -> msg.matches("should replace [a-zA-Z0-9\\.]+\\\\(fmt.Sprintf\\\\(...\\\\)\\\\) with [a-zA-Z0-9]+.Errorf\\\\(...\\\\)")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource can be simplified `contains("(fmt.Sprintf(...))") && contains(".Errorf(...)")`

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 8]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("Errors", msg -> msg.matches("error var [a-zA-Z0-9]+ should have name of the form [a-zA-Z0-9]+")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource let's avoid regexp here (starts and contains)

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 9]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("ErrorStrings", msg -> msg.startsWith("error strings should not be capitalized or end with punctuation or a newline")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource `equals`

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 10]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("ReceiverNames", msg ->
+      msg.contains("should be consistent with previous receiver name") ||
+        msg.startsWith("receiver name should not be an underscore") ||
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: @andrea-guarino-sonarsource why not full message with equals?

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 11]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
-    new ExternalKey("SpecificSuffix", msg -> msg.contains("don't use unit-specific suffix")),
-    new ExternalKey("StructFieldInitialisms", msg -> msg.matches("struct field [a-zA-Z0-9]+ should be [a-zA-Z0-9]+")),
-    new ExternalKey("UnderscoreInGoName", msg -> msg.startsWith("don't use underscores in Go names;")),
-    new ExternalKey("UnderscoreInPackageName", msg -> msg.startsWith("don't use an underscore in package name"))
+    new ExternalKey("Names", msg ->
+      msg.startsWith("don't use an underscore in package name") ||
+        msg.startsWith("don't use ALL_CAPS in Go names; use CamelCase") ||
+        msg.startsWith("don't use leading k in Go names;") ||
+        msg.startsWith("don't use underscores in Go names;") ||
+        msg.matches("(range var|struct field|[a-zA-Z0-9]+) [a-zA-Z0-9]+ should be [a-zA-Z0-9]+") ||
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: fix it

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java [hunk 12]
```diff
@@ -58,43 +58,43 @@ private ExternalKeyUtils() {
...
+    new ExternalKey("Elses", msg -> msg.contains("if block ends with a return statement, so drop this else and outdent its block")),
+    new ExternalKey("Ranges", msg -> msg.matches("should omit (2nd value|values) from range; this loop is equivalent to .*")),
+    new ExternalKey("Errorf", msg -> msg.matches("should replace [a-zA-Z0-9\\.]+\\\\(fmt.Sprintf\\\\(...\\\\)\\\\) with [a-zA-Z0-9]+.Errorf\\\\(...\\\\)")),
+    new ExternalKey("Errors", msg -> msg.matches("error var [a-zA-Z0-9]+ should have name of the form [a-zA-Z0-9]+")),
+    new ExternalKey("ErrorStrings", msg -> msg.startsWith("error strings should not be capitalized or end with punctuation or a newline")),
+    new ExternalKey("ReceiverNames", msg ->
+      msg.contains("should be consistent with previous receiver name") ||
+        msg.startsWith("receiver name should not be an underscore") ||
+        msg.equals("receiver name should be a reflection of its identity; don't use generic names such as \"this\" or \"self\"")),
+    new ExternalKey("IncDec", msg -> msg.matches("should replace [a-zA-Z0-9\\-\\+=\\.\\(\\) ]+ with [a-zA-Z0-9\\-\\+\\.\\(\\)]+")),
```
Comment (vilchik-elena) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/externalreport/ExternalKeyUtils.java]: fix it

**Review by vilchik-elena — CHANGES_REQUESTED** [review]

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**